### DISCUSSION TITLE
Can't delete a file

### DISCUSSION THREAD

**Discussion by santsang** [community/community#50441] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

I tried to delete a file. The bin icon is dim (action is not allowed).  When I click on it, it says: you must be on a branch...  Thank you very much for your suggestion.  Sandro

**Comment by MohamedYansane** [comment]
type git branch to see if your main branch is selected you'll see * for the branch which is selected. if your main branch is not selected just type git switch yourbranchname

**Reply by santsang** [reply]
Thanks MohamedYansane for your suggestion.  All in a sudden, github decided to let me to delete it without problem.  However, I am interested in the method you proposed.  It will be useful for the long run.  Would you mind showing me the screenshots?  For example, loads of blogs suggest rename a folder using web interface like this: https://stackoverflow.com/questions/31861651/how-to-rename-a-directory-folder-on-github-website.  It did not work for me at all.  Many thanks for your help.  Sandro

**Reply by MohamedYansane** [reply]
before renaming a file just a trick try to do changes inside the file or folder or even track the file . In my case on my src folder i have this directory src/notest.txt so after making changes inside it before committing it. I just checked if it's tracked if it's i'll copy the file path from my git status then type the command:
git mv src/notest.txt src/notes.txt . After typing that command i can commit now then  type git status

**Reply by santsang** [reply]
Many thanks for the discussion.  Yes, I also find that command line is much easier to understand.  Sandro

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @santsang ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!

**Reply by santsang** [reply]
Many thanks.  We are users.  Mostly we are exhausted by the problem when we post...  We don't have the energy to look up the guidelines... in order to ask for help.  Anyway, I am glad that you did not ask me to do this and that to move the post...
